SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ExpectedViewDelta.cpp

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "ComponentTestUtils.h"
+#include "SpatialView/ViewDelta.h"
+#include "Tests/SpatialView/ExpectedEntityDelta.h"
+#include "Tests/SpatialView/ExptectedViewDelta.h"
+
+using namespace SpatialGDK;
+
+ExpectedViewDelta& ExpectedViewDelta::AddEntityDelta(Worker_EntityId EntityId, EntityChangeType Status)
+{
+	EntityDeltas.Add(EntityId, { EntityId, Status == ADD, Status == REMOVE });
+	return *this;
+}
+
+ExpectedViewDelta& ExpectedViewDelta::AddComponentAdded(Worker_EntityId EntityId, ComponentData ComponentData)
```

Comment:
Nit - naming an argument the same thing as the type can cause problems, as the second definition of the symbol shadows the first.
It's a weird thing where it's ill-formed but compilers aren't obliged to complain. So MSVC and clang don't but gcc does.
The underlying issue is that I should have called `ComponentData` `FComponentData` or something. In the repo I made a lot of this stuff the naming conventions for fields never overlapped with types so I didn't think about it too much.